modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/tests/MultipleSelect.spec.tsx

```diff
@@ -0,0 +1,311 @@
+							{
+								_key: 'Organization Administrator',
+								checked: true,
+								label: 'Organization Administrator',
+								value: 'Organization Administrator',
+							},
+						],
+						label: 'Organization Roles',
+						value: 'organizationRoles',
+					},
+				])
+			}
+		/>
+	);
+
+	fireEvent.click(screen.getByRole('combobox'));
+
+	fireEvent.click(screen.getByText('Account Administrator'));
+
+	expect(screen.getByLabelText('select-all')).not.toBeChecked();
+});
+
+it('search an option', () => {
+	render(
+		<MultipleSelect
+			disabled={false}
+			id="multipleSelect"
+			options={mockOptions}
+			search
+			searchPlaceholder="Search for an option"
+			selectAllOption
+			setOptions={() => {}}
+		/>
+	);
+
+	fireEvent.click(screen.getByRole('combobox'));
+	fireEvent.change(screen.getByPlaceholderText('Search for an option'), {
+		target: {value: 'Account'},
+	});
+
+	expect(screen.getByText('Account Administrator')).toBeVisible();
+	expect(screen.getByText('Account Member')).toBeVisible();
+	expect(screen.getByText('Account Manager')).toBeVisible();
+
+	expect(screen.getAllByText('Administrator')).toHaveLength(1);
+	expect(screen.getAllByText('Analytics Administrator')).toHaveLength(1);
+	expect(screen.getAllByText('Organization Administrator')).toHaveLength(1);
+});
+
+it('unchecked all options using unselect all', () => {
+	render(
+		<MultipleSelect
+			disabled={false}
+			id="multipleSelect"
+			options={mockOptions}
+			selectAllOption
+			setOptions={(options) =>
+				expect(options).toStrictEqual([
+					{
+						children: [
+							{
+								_key: 'Account Administrator',
+								checked: false,
+								label: 'Account Administrator',
+								value: 'Account Administrator',
+							},
+							{
+								_key: 'Account Member',
+								checked: false,
+								label: 'Account Member',
+								value: 'Account Member',
+							},
+						],
+						label: 'Account Roles',
+						value: 'accountRoles',
+					},
+					{
+						children: [
+							{
+								_key: 'Administrator',
+								checked: false,
+								label: 'Administrator',
+								value: 'Administrator',
+							},
+							{
+								_key: 'Analytics Administrator',
+								checked: false,
+								label: 'Analytics Administrator',
+								value: 'Analytics Administrator',
+							},
+						],
+						label: 'Regular Roles',
+						value: 'regularRoles',
+					},
+					{
+						children: [
+							{
+								_key: 'Account Manager',
+								checked: false,
+								label: 'Account Manager',
+								value: 'Account Manager',
+							},
+							{
+								_key: 'Organization Administrator',
+								checked: false,
+								label: 'Organization Administrator',
+								value: 'Organization Administrator',
+							},
+						],
+						label: 'Organization Roles',
+						value: 'organizationRoles',
+					},
+				])
+			}
+		/>
+	);
+
+	fireEvent.click(screen.getByTitle('Clear All'));
+
+	expect(screen.getByLabelText('select-all')).not.toBeChecked();
```

Comment:
don't you think is also necessary to see if the options are not checked as well? because if the `select all` is not working correctly, it could be unchecked while the other options are still checked (this being a bug that the test would catch). make sense to you or only this verification here is enough?

Reply:
Hey Rebeca, as I'm testing the data structure that will be returned with expect().toStrictEqual(), if the data structure is correct, ClayCheckbox will internally be responsible for checking and unchecking the options based on the data structure. data.

Therefore, I see no need to do this check. I think I would be testing Clay's internal logic.

As I couldn't access the selectAllChecked state outside the component, I decided to assert that it would not be marked by the UI and not by the data structure, but as a matter of limitation.

In this case I am testing the actions performed in this [onItemsChange](https://github.com/liferay/liferay-portal/blob/4da7307aa24e47b212e231802018cd7ced24bd70/modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/MultipleSelect.tsx#L132-L154).

Reply:
makes sense, thanks for explaining!